libraries/chain/database.cpp

```diff
@@ -123,7 +123,8 @@ void database::open( const fc::path& data_dir, const fc::path& shared_mem_dir, u
             undo_all();
             FC_ASSERT( revision() == head_block_num(), "Chainbase revision does not match head block num",
                ("rev", revision())("head_block", head_block_num()) );
-            validate_invariants();
+            if (_validate_invariants)
```

Comment:
I'm not sure I like using an instance variable for this when it is only used in this one function. I think adding it is a parameter to open with default initialization to false makes more sense.